app/src/main/java/id_iot/demo/CollectionAdapter.java

```diff
@@ -0,0 +1,102 @@
+package id_iot.demo;
+
+import android.content.Context;
+import android.support.v7.widget.SearchView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.Filterable;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Filter;
+import android.widget.Filterable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CollectionAdapter extends BaseAdapter implements Filterable {
+    private Context mContext;
+    private ArrayList<Recipe> recipes;
+    private ArrayList<Recipe> backupRecipes;
+    ValueFilter valueFilter;
+
+    public CollectionAdapter(Context c, ArrayList<Recipe> recipes) {
+        mContext = c;
+        this.recipes = recipes;
+        this.backupRecipes = recipes;
+    }
+
+
+    @Override
+    public long getItemId(int position) {
+        return position;
+    }
+
+    @Override
+    public Object getItem(int position) {
+        return recipes.get(position);
+    }
+
+    @Override
+    public int getCount() {
+        return recipes.size();
+    }
+
+    public View getView(int position, View convertView, ViewGroup parent) {
+        final LayoutInflater layoutInflater = LayoutInflater.from(mContext);
```

Comment:
for some reason you used 'final' modifier here. It is actually a really good practise to have immutable variables. But the rest of your code is missing this. Why?